postman-generator/src/main/scala/generator/DependantOperationResolver.scala

```diff
@@ -0,0 +1,208 @@
+      .get(key = objRefAttr.relatedServiceNamespace) match {
+      case Some(resources) =>
+        val operations =
+          resources
+            .filter(_.`type` === objRefAttr.resourceType)
+            .flatMap(_.operations)
+
+        val referencedOperationOpt = findOperationThatEndsWithOrEquals(operations, objRefAttr.operationMethod, objRefAttr.operationPath)
+
+        val deleteOperationOpt = for {
+          deleteOpPath <- objRefAttr.deleteOperationPath.map(_.toLowerCase)
+          deleteOp <- findOperationThatEndsWithOrEquals(operations, Method.Delete, deleteOpPath)
+        } yield deleteOp
+
+        referencedOperationOpt.map { referencedOperation =>
+          DependantOperations(referencedOperation, deleteOperationOpt)
+        }
+      case None =>
+        logger.warn(s"ResolvedService namespace to resources map (includes imported services) does not contain key = ${objRefAttr.relatedServiceNamespace} / Can't find the referenced operation for $objRefAttr")
+        None
+    }
+  }
+
+  private def findOperationThatEndsWithOrEquals(operations: Seq[Operation], method: Method, pathSuffix: String): Option[Operation] = {
+    operations.find { o =>
+      o.method === method &&
+        (o.path.toLowerCase.endsWith(pathSuffix.toLowerCase()) || o.path.equalsIgnoreCase(pathSuffix))
+    }
+  }
+
+  private def deepSearchModelsForAttributes(typ: String, service: Service): Seq[ObjectReference] = {
+
+    def recurSearch(typ: String): Seq[ObjectReference] = {
+      service.models.find(_.name === typ) match {
+        case Some(model) =>
+          model.fields.flatMap {
+            // model have fields with another models, going deeper
+            case field if service.models.exists(_.name === field.`type`) =>
+              recurSearch(field.`type`)
+            // model is a "leaf", searching for special attribute
+            case field =>
+              val objRefAttrOpt = findReferenceAttributeInModelField(field)
+
+              objRefAttrOpt match {
+                case Some(objAttrRef) =>
+                  val modelToLookup =
+                    if (objAttrRef.relatedServiceNamespace != service.namespace)
+                      s"${objAttrRef.relatedServiceNamespace}.models.${objAttrRef.resourceType}"
+                    else
+                      objAttrRef.resourceType // namespace from the main service, plain model name
+                  // adding and going recursive
+                  recurSearch(modelToLookup) :+ objAttrRef
+                case None =>
+                  Nil
+              }
+          }
+        case None if service.unions.exists(_.name === typ) =>
+          val union = service.unions.find(_.name === typ).get
+          union.types.flatMap(u => recurSearch(u.`type`))
+        case _ =>
+          Nil
+      }
+    }
+
+    recurSearch(typ)
+  }
+
+  private def findParameterInPathString(path: String): Option[Parameter] = {
+    val regex = """\:(\w+)[\/]{0,1}""".r
```

Comment:
This regex appears `"\:(\w+)[\/]{0,1}` 2x -> please move that to some object

Reply:
extracted